database/mssql/scripts/dbscripts/PSP_PIMS_S25_01/Alter Down/88_DML_PIMS_ROLES_CLAIMS.sql

```diff
@@ -0,0 +1,87 @@
+/* -----------------------------------------------------------------------------
+Delete all data from the PIMS_ROLE_CLAIM table and repopulate.
+. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
+Author        Date         Comment
+------------  -----------  -----------------------------------------------------
+Devin Smith  2021-Nov-18  Initial version
+----------------------------------------------------------------------------- */
+
+DECLARE @admin BIGINT;
+DECLARE @rem BIGINT;
+DECLARE @rea BIGINT;
+SELECT @admin = ROLE_ID FROM PIMS_ROLE where NAME = 'System Administrator';
```

Comment:
@dfilteau it seems like the order of roles was changed when a new row was inserted at some point, This was leading to a number of roles not having the correct claims associated to it. Hopefully by getting the correct ID explicitly here that will not happen again in the future.

My question is, do I need to do this with the PIMS_CLAIM table as well? currently this script is depending on all of the claims ids to be static. If they were ever re-ordered this script would assign incorrect claims to roles.

Reply:
Hi @devinleighsmith, I made no changes to the order of the roles so I'm not sure where this change came from.  If you could provide the roles in the correct order, I'd appreciate it.

Reply:
This PR addresses the issue.

The issue appears to have been introduced when the value UNDETERMINED was introduced that value was introduced before the pre-existing roles of Real Estate Manager and Real Estate Analyst

![image](https://user-images.githubusercontent.com/41091511/163078467-c687f8ca-6e66-417e-b404-dd5a797701f0.png)